src/curl.cpp

```diff
@@ -123,9 +132,18 @@ namespace powerloader
         return *this;
     }
 
-    CURLHandle& CURLHandle::url(const std::string& url)
+    CURLHandle& CURLHandle::url(const std::string& url, const proxy_map_type& proxies)
```

Comment:
Do we want to force the user to pass a proxy map? Or do we still have use cases where it makes sense to call `url` with a string only?